roles/container-engine/docker/tasks/main.yml

```diff
@@ -46,7 +46,7 @@
           docker requires a minimum kernel version of
           {{ docker_kernel_min_version }} on
           {{ ansible_distribution }}-{{ ansible_distribution_version }}
-  when: (not ansible_os_family in ["CoreOS", "Container Linux by CoreOS", "ClearLinux"]) and (ansible_kernel is version(docker_kernel_min_version, "<"))
+  when: (not ansible_os_family in ["CoreOS", "Coreos", "Container Linux by CoreOS", "ClearLinux"]) and (ansible_kernel is version(docker_kernel_min_version, "<"))
```

Comment:
why CoreOS + Coreos? You didn't do this for other checks

Reply:
I tried to change it at every location. Depending on the Ansible version CoreOS can be detected as `CoreOS`, `Coreos`, or `Container Linux by CoreOS`. The `Coreos` thing was introduced in 2.8 or 2.9 when they did a refactor around OS detection lib.

It might be time to consider a `is_coreos` var to abstract all this mess :)

Reply:
I'm still working on fixing this mess. Sorry about that. I'd much rather get this correct in Ansible rather than having you work around it.

I got stuck trying to sort out 1) What _should_ the OS family string actually be and 2) Getting adequate samples of the various CoreOS flavors to test against.

Last I looked there was Container Linux by CoreOS, CoreOS, Fedora CoreOS, and Red Hat Enterprise Linux CoreOS. At the time Fedora CoreOS didn't have builds I could test with. Maybe there are things are a bit more concrete now that RHEL 8 has shipped.

Reply:
While investigating this issue, I've found this PR that seems to have introduced the change: https://github.com/ansible/ansible/pull/52199. In short, on newer Ansibles, the value of `ansible_os_familiy` will be fetched by the ID inside the `/etc/os-release` file instead of `name` as it was previously. Since CoreOS is [reaching its end of life](https://coreos.com/os/eol/) soon, I would suggest leave the fix as it is and when Fedora CoreOS is finally released, it should be included in the "reliable" list of OS ids inside the Ansible project itself.